### PR TITLE
Goodbye Kord, Welcome JDA

### PR DESCRIPTION
**Pull request by Johnny850807**
## Why need this change? / Root cause: 
- Kord sucks
## Changes made:
- Remove all Kord/DiscordKt related dependencies and use JDA in all modules
## Test Scope / Change impact:
- Landing-Experience

### CODE REVIEW

File: discord-impl-jda/src/main/kotlin/tw/waterballsa/utopia/jda/jda.kt [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+internal fun loadListenersFromAllUtopiaModules(wsa: WsaDiscordProperties?): List<UtopiaListener> {
+
+    val listeners = mutableListOf<UtopiaListener>()
+
+    val reflections = Reflections("tw.waterballsa.utopia", SubTypes, TypesAnnotated, MethodsReturn)
+    val listenerFunctions = reflections.get(MethodsReturn.with(UtopiaListener::class.java).`as`(Method::class.java))
+            .filterNot {
+                it.declaringClass.`package`.name.startsWith("tw.waterballsa.utopia.jda")
+            }
+            .toList()
```
Comment (akechetaki10736) [inline discord-impl-jda/src/main/kotlin/tw/waterballsa/utopia/jda/jda.kt:88]: filterNot 本身就回傳 List，後面不用再加 toList

File: commons/src/main/kotlin/tw/waterballsa/utopia/commons/config/WsaDiscordProperties.kt [hunk 2]
```diff
@@ -1,34 +1,26 @@
...
-        unlockEntryMessageId = properties.getProperty("unlock-entry-message-id").toULong()
-        selfIntroChannelId = properties.getProperty("self-intro-channel-id").toULong()
-        wsaGuestRoleId = properties.getProperty("wsa-guest-role-id").toULong()
-        wsaCitizenRoleId = properties.getProperty("wsa-citizen-role-id").toULong()
+        guildId = properties.getProperty("guild-id")
+        unlockEntryMessageId = properties.getProperty("unlock-entry-message-id")
+        selfIntroChannelId = properties.getProperty("self-intro-channel-id")
+        wsaGuestRoleId = properties.getProperty("wsa-guest-role-id")
+        wsaCitizenRoleId = properties.getProperty("wsa-citizen-role-id")
     }
```
Comment (akechetaki10736) [inline commons/src/main/kotlin/tw/waterballsa/utopia/commons/config/WsaDiscordProperties.kt:24]: 可以考慮

```kotlin
properties.run {
    guildId = getProperty("guild-id")
    unlockEntryMessageId = getProperty("unlock-entry-message-id")
    selfIntroChannelId = getProperty("self-intro-channel-id")
    wsaGuestRoleId = getProperty("wsa-guest-role-id")
    wsaCitizenRoleId = getProperty("wsa-citizen-role-id")
}
```

**Review by hokou — APPROVED** [review]
LGTM

File: landing-experience/src/main/kotlin/tw/waterballsa/utopia/landingx/ReactionRoles.kt [hunk 3]
```diff
@@ -1,71 +1,72 @@
...
 ) {
     if (roles.isEmpty()) {
         return
     }
     roles.forEach { role ->
-        discord.kord.rest.guild.deleteRoleFromGuildMember(guildId, user.id, role)
+        guild.removeRoleFromMember(user, role)
+                .queue {
+                    logger.info { "[Role removed] {\"userName\":\"${user.name}\", \"roleId\":\"${role.id}\" }" }
+                }
```
[the remark this answers is not in the document]
Comment (Johnny850807) [reply landing-experience/src/main/kotlin/tw/waterballsa/utopia/landingx/ReactionRoles.kt:70]: thanks

**Pull request merged**